js/charts/heatmap.js

```diff
@@ -637,58 +635,56 @@ function heatmap() {
+                        rect.classed('selected', false);
+                    } else {
+                        rect.classed('selected', true);
+                    }
 
-                        var _filterDimension = {};
-                        if (broadcast.filterSelection.id) {
-                            _filterDimension = broadcast.filterSelection.filter;
-                        } else {
-                            broadcast.filterSelection.id = parentContainer.attr('id');
-                        }
-                        var dimension = _dimension[0];
-                        if (_filterDimension[dimension]) {
-                            var temp = _filterDimension[dimension];
-                            if (temp.indexOf(d.y) < 0) {
-                                temp.push(d.y);
-                            } else {
-                                temp.splice(temp.indexOf(d.y), 1);
-                            }
-                            _filterDimension[dimension] = temp;
+                    var _filterDimension = {};
+                    if (broadcast.filterSelection.id) {
+                        _filterDimension = broadcast.filterSelection.filter;
+                    } else {
+                        broadcast.filterSelection.id = parentContainer.attr('id');
+                    }
+                    var dimension = _dimension[0];
+                    if (_filterDimension[dimension]) {
+                        var temp = _filterDimension[dimension];
+                        if (temp.indexOf(d.y) < 0) {
+                            temp.push(d.y);
                         } else {
-                            _filterDimension[dimension] = [d.y];
+                            temp.splice(temp.indexOf(d.y), 1);
                         }
-
-                        _filterDimension[dimension]._meta = {
-                            dataType: _dimensionType[0],
-                            valueType: 'castValueType'
-                        };
-
-                        var idWidget = broadcast.updateWidget[parentContainer.attr('id')];
-                        broadcast.updateWidget = {};
-                        broadcast.updateWidget[parentContainer.attr('id')] = idWidget;
-                        broadcast.filterSelection.filter = _filterDimension;
-                        var _filterParameters = filterParameters.get();
-                        _filterParameters[dimension] = _filterDimension[dimension];
-                        filterParameters.save(_filterParameters);
+                        _filterDimension[dimension] = temp;
+                    } else {
+                        _filterDimension[dimension] = [d.y];
                     }
+
+                    _filterDimension[dimension]._meta = {
+                        dataType: _dimensionType[0],
+                        valueType: 'castValueType'
+                    };
+
+                    var idWidget = broadcast.updateWidget[parentContainer.attr('id')];
```

Comment:
looks like these lines of code is being used in all charts...pls keep it as util function in separate file

                    var idWidget = broadcast.updateWidget[parentContainer.attr('id')];
                    broadcast.updateWidget = {};
                    broadcast.updateWidget[parentContainer.attr('id')] = idWidget;
                    broadcast.filterSelection.filter = _filterDimension;
                    var _filterParameters = filterParameters.get();
                    _filterParameters[dimension] = _filterDimension[dimension];
                    filterParameters.save(_filterParameters);